deployments/development-install/helmfile.yaml

```
@@ -0,0 +1,187 @@
+releases:
+  - name: ocis
+    chart: ../../charts/ocis
+    namespace: ocis
+    values:
+      - externalDomain: ocis.kube.owncloud.test
+      - ingress:
+          enabled: true
+          ingressClassName: nginx
+          annotations:
+            nginx.ingress.kubernetes.io/proxy-body-size: 1024m
+
+      - logging:
+          level: debug
+
+      - insecure:
+          oidcIdpInsecure: true
+          ocisHttpApiInsecure: true
+
+      - services:
+          idm:
+            persistence:
+              enabled: true
+              chownInitContainer: true
+              accessModes:
+                - ReadWriteOnce
+
+          nats:
+            persistence:
+              enabled: true
+              chownInitContainer: true
+              accessModes:
+                - ReadWriteOnce
+
+          search:
+            persistence:
+              enabled: true
+              chownInitContainer: true
+              accessModes:
+                - ReadWriteOnce
+
+          storagesystem:
+            persistence:
+              enabled: true
+              chownInitContainer: true
+              accessModes:
+                - ReadWriteOnce
+
+          storageusers:
+            persistence:
+              enabled: true
+              chownInitContainer: true
+              accessModes:
+                - ReadWriteOnce
+
+          store:
+            persistence:
+              enabled: true
+              chownInitContainer: true
+              accessModes:
+                - ReadWriteOnce
+            extraLabels:
+              foo: bar
+
+          thumbnails:
+            persistence:
+              enabled: true
+              chownInitContainer: true
+              accessModes:
+                - ReadWriteOnce
```

Maybe an iteration might be better:

```suggestion
          {{ range tuple "idm" "nats" "search" "storagesystem" "storageusers" "store" "thumbnails" }}
          {{ . }}:
            persistence:
              enabled: true
              chownInitContainer: true
              accessModes:
                - ReadWriteOnce
          {{- end }}
```

---

While this is cleaner, it also becomes very annoying if someone wants to change/add settings in those services, no?

---

Yeah that's true. You could add another layer to solve this by having a values.yaml with

```YAML
services:
  additionals:
    store:
      extraLabels:
        foo: "bar"
```

And then 

```YAML
          {{ range tuple "idm" "nats" "search" "storagesystem" "storageusers" "store" "thumbnails" }}
          {{ . }}:
            persistence:
              enabled: true
              chownInitContainer: true
              accessModes:
                - ReadWriteOnce
            {{- if (index $.Values.services.additionals .) }}
{{ toYaml (index $.Values.services.additionals .) | indent 12 }}
            {{- end }}
          {{- end }}
```

It just depends if you consider, that people should modify the helmfile.yaml directly or if they should modify a values file. I think there is no rule of thumb.

---

I'd go for leaving it like this, because as these are examples, a very normal workflow would be to copy them and edit them.

---

Ok